src/network/network.cpp

```diff
@@ -1424,9 +1462,49 @@ void Network::Server_Handle_TOKEN(NetworkConnection& connection, NetworkPacket&
 	for (int i = 0; i < token_size; i++) {
 		connection.Challenge[i] = (uint8)(rand() & 0xff);
 	}
+	rct_object_entry object_entries[OBJECT_ENTRY_COUNT];
+	int count = scenario_get_num_packed_objects_to_write(object_entries);
+	Server_Send_OBJECTS(connection, object_entries, count);
```

Comment:
I know, but it was much easier for me to code. Otherwise I'll need to add another roundtrip after authentication and before sending the map, but I can probably fix it properly.